### PR TITLE
MDEV-34340 netbsd fix crash on MariaDB-backup

### PR DESCRIPTION
**Pull request by grooverdan**
- [x] *The Jira issue number for this PR is: MDEV-34340*

## Description

Without a get_exe implementation there was a null pointer exception my_realpath.

Refactor to mysys library as requested by @vuvova.

## Release Notes

Fix NetBSD compatibility in MariaDB backup thanks to Thomas Klausner

## How can this PR be tested?

Confirmed by contributor. (or we setup a NetBSD test)

## Basing the PR against the correct MariaDB version
- [ ] *This is a new feature or a refactoring, and the PR is based against the `main` branch.*
- [X] *This is a bug fix, and the PR is based against the earliest maintained branch in which the bug can be reproduced.*

## PR quality check
- [X] I checked the [CODING_STANDARDS.md](https://github.com/MariaDB/server/blob/-/CODING_STANDARDS.md) file and my PR conforms to this where appropriate.
- [X] For any trivial modifications to the PR, I am ok with the reviewer making the changes themselves.

### CODE REVIEW

**Comment by vuvova** [comment]
I thought it should also replace similar code in `mysys/my_init.c` — and _that's why_ this function could be in mysys, because it's universal enough to be used in at least two places.

File: mysys/my_getexe.c [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/* Copyright (c) 2017, MariaDB Plc
```
Comment (LinuxJedi) [inline mysys/my_getexe.c:1]: Is it though?

**Review by LinuxJedi — COMMENTED** [review]
One comment in the review.

Also, if Sergei is referring to the symlink check in `my_init`. With recent discussions that might end up going away anyway.

File: mysys/my_getexe.c (hunk 1 above)
Comment (grooverdan) [reply mysys/my_getexe.c:1]: Vlad was the plc author of the entire my_get_exepath function that was copied here and its authored year of 2017 was used.

**Comment by grooverdan** [comment]
> I thought it should also replace similar code in `mysys/my_init.c` — and _that's why_ this function could be in mysys, because it's universal enough to be used in at least two places.

Regardless, this is a 10.5 fix. To adjust my_init.c for checking replacing a symlink check would be in the 11.x branch (if discussions of its obsoletes got ahead). Regardless it becomes a common library to use.

**Pull request merged**